rust/basic_bitcoin/src/basic_bitcoin/src/ecdsa_api.rs

```diff
@@ -0,0 +1,53 @@
+use crate::types::*;
+use ic_cdk::{call, export::Principal};
+
+const KEY_NAME: &str = "test";
+
+/// Returns the ECDSA public key of this canister at the given derivation path.
+pub async fn ecdsa_public_key(derivation_path: Vec<Vec<u8>>) -> Vec<u8> {
+    // TODO: Replace this principal with the management canister when it's available.
+    // For now, call a canister that provides a mock implementation.
+    let ecdsa_canister_id = Principal::from_text("ryjl3-tyaaa-aaaaa-aaaba-cai").unwrap();
+
+    // Retrieve the public key of this canister at derivation path [0]
+    // from the ECDSA API.
+    let res: (ECDSAPublicKeyReply,) = call(
+        ecdsa_canister_id,
+        "ecdsa_public_key",
+        (ECDSAPublicKey {
+            canister_id: None,
+            derivation_path,
+            key_id: EcdsaKeyId {
+                curve: EcdsaCurve::Secp256k1,
+                name: String::from(KEY_NAME),
+            },
```

Comment:
Maybe having a small function returning the `key_id` used for both `ecdsa_public_key` and `sign_with_ecdsa` would simplify a bit the code.

Reply:
That part is subjective. I personally prefer in-lining when it's this small, since it's clearer to see everything and to copy for developers than having to look up another method.